### PR TITLE
Filter out existing cases from string completions

### PR DESCRIPTION
**Pull request by Andarist**
fixes https://github.com/microsoft/TypeScript/issues/52614

cc @gabritto @andrewbranch

### CODE REVIEW

File: src/services/stringCompletions.ts [hunk 1]
```diff
@@ -276,6 +288,8 @@ function stringLiteralCompletionDetails(name: string, location: Node, completion
         }
         case StringLiteralCompletionKind.Types:
             return find(completion.types, t => t.value === name) ? createCompletionDetails(name, ScriptElementKindModifier.none, ScriptElementKind.typeElement, [textPart(name)]) : undefined;
+        case StringLiteralCompletionKind.Cases:
+            return;
```
Comment (Andarist) [inline src/services/stringCompletions.ts:292]: I wasn't sure what kind of details should be returned here.

File: src/services/completions.ts [hunk 2]
```diff
@@ -1105,75 +1100,6 @@ function getExhaustiveCaseSnippets(
     return undefined;
 }
 
-interface CaseClauseTracker {
-    addValue(value: string | number): void;
-    hasValue(value: string | number | PseudoBigInt): boolean;
-}
-
-function newCaseClauseTracker(checker: TypeChecker, clauses: readonly (CaseClause | DefaultClause)[]): CaseClauseTracker {
```
Comment (Andarist) [inline src/services/completions.ts:1113]: I think it makes sense to move this out to a shared space so both `completions` and `stringCompletions` can use it - but if you prefer some other solution, let me know.

File: src/services/stringCompletions.ts (hunk 1 above)
Comment (gabritto) [reply src/services/stringCompletions.ts:292]: I think you should do something like the above case for `StringLiteralCompletionKind.Types`, except I don't think the `kind` should be `ScriptElementKind.typeElement`, it should be `ScriptElementKind.string` instead. Not related to your PR: I don't know why the call above is using `ScriptElementKind.typeElement`, it looks weird to me that the `kind` in the `CompletionEntry` and `CompletionEntryDetails` return two different things for `StringLiteralCompletionKind.Types` case, do you know why that is @@andrewbranch.

Comment (Andarist) [reply src/services/stringCompletions.ts:292]: I came to the conclusion - that perhaps I don't need a completely separate type and I can just use `.Types`. At first, I wasn't sure what `.Types` is supposed to resolve so I was afraid of reusing it. WDYT?

File: tests/cases/fourslash/switchCompletions.ts [hunk 3]
```diff
@@ -13,6 +13,12 @@
 ////         return 1;
 ////     case /*2*/
 //// }
+//// declare const f2: 'foo' | 'bar' | 'baz';
+//// switch (f) {
```
[the remark this answers is not in the document]
Comment (Andarist) [reply tests/cases/fourslash/switchCompletions.ts:17]: yes, definitely - this was a late copy-paste from a separate file in which I was originally testing it 😅 gonna amend that soon

File: src/services/stringCompletions.ts (hunk 1 above)
Comment (andrewbranch) [reply src/services/stringCompletions.ts:292]: I don’t know the answer to @gabritto’s question, but I do think you can just use `.Types` for this.

Comment (gabritto) [reply src/services/stringCompletions.ts:292]: I think you can reuse it and fix the `CompletionEntryDetails.kind` weirdness. With any luck, if this turns out to be wrong for some reason, a test will break 😅

**Comment by jakebailey** [comment]
@Andarist FYI you may want to recopy the settings from `settings.template.json` to get the organize import settings in the repo (+ use the nightly TS extension), which should make thinkgs like auto-imports produce a more linter-friendly result.

**Review by gabritto — APPROVED** [review]

**Pull request merged**